pkg/network/ebpf/c/protocols/classification/usm-context.h

```diff
@@ -0,0 +1,82 @@
+#ifndef __USM_CONTEXT_H
+#define __USM_CONTEXT_H
+
+#include "tracer.h"
+#include "protocols/classification/common.h"
+#include "protocols/classification/defs.h"
+#include "protocols/classification/maps.h"
+
+// from uapi/linux/if_packet.h
+#define PACKET_OUTGOING 4
+
+typedef struct {
+    char data[CLASSIFICATION_MAX_BUFFER];
+    size_t size;
+} classification_buffer_t;
+
+typedef struct {
+    struct __sk_buff *owner;
+    conn_tuple_t tuple;
+    skb_info_t  skb_info;
+    classification_buffer_t buffer;
+} usm_context_t;
+
+// Kernels before 4.7 do not know about per-cpu array maps.
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 7, 0)
+
+// A per-cpu buffer used to read requests fragments during protocol
+// classification and avoid allocating a buffer on the stack. Some protocols
+// requires us to read at offset that are not aligned. Such reads are forbidden
+// if done on the stack and will make the verifier complain about it, but they
+// are allowed on map elements, hence the need for this map.
+BPF_PERCPU_ARRAY_MAP(classification_buf, __u32, usm_context_t, 2)
```

Comment:
This has to do with the eBPF preemption discussion we had. Instead of using one entry per CPU we use 2 to untangle ingress and egress socket filter execution. I'll add add a note to clarify that

Reply:
Added a note